frontend/src/components/dockerDesktopSetup/index.tsx

```diff
@@ -0,0 +1,134 @@
+import { Icon } from '@iconify/react';
+import { Box } from '@material-ui/core';
+import Button from '@material-ui/core/Button';
+import Card from '@material-ui/core/Card';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import MobileStepper from '@material-ui/core/MobileStepper';
+import Paper from '@material-ui/core/Paper';
+import { makeStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import { useEffect, useState } from 'react';
+import { generatePath } from 'react-router';
+import { Redirect } from 'react-router-dom';
+import { useHistory } from 'react-router-dom';
+import helpers from '../../helpers';
+import { refreshConfig } from '../../lib/k8s';
+import { getClusterPrefixedPath } from '../../lib/util';
+import { ClusterDialog } from '../cluster/Chooser';
+import { DialogTitle } from '../common/Dialog';
+
+const useStyle = makeStyles({
+  card: {
+    padding: '1rem 2rem',
+  },
+  image: {
+    width: '90%',
+  },
+});
+
+export function DockerDesktopSetup() {
+  const classes = useStyle();
+  const history = useHistory();
+  const [authenticated, setAuthenticated] = useState(false);
+
+  async function refreshConfigAndTestAuth() {
+    const config = await refreshConfig();
+    if (config && config['clusters'].length > 0) {
+      helpers.setRecentCluster(config['clusters'][0]);
+      history.push({
+        pathname: generatePath(getClusterPrefixedPath(), {
+          cluster: config['clusters'][0].name,
+        }),
+      });
+      setAuthenticated(true);
+    }
+  }
+
+  useEffect(() => {
```

Comment:
Refreshing config would be useful for the app/local version of headlamp too. This is a long running annoying thing for me.

I'm wondering if we enable the refresh config behaviour for all Headlamp?

Reply:
Makes sense, Created a new issue for the same
https://github.com/kinvolk/headlamp/issues/850